ext/EnzymeExt.jl

```diff
@@ -253,7 +255,7 @@ function gpu_rev(
 end
 
 function EnzymeRules.augmented_primal(
-        config::Config,
+        config::RevConfig,
         func::Const{<:Kernel},
```

Comment:
Do we need those kinds of changes with an abstract Config type?

Reply:
Forward mode rules changed fairly fundamentally, so we'll definitely still need these changes